[wip] Ref/design system/workspace

seaerchin opened the pull request:
## Problem

<!-- What problem are you trying to solve? What issue does this close? -->

Closes [insert issue #]

## Solution

<!-- How did you solve the problem? -->

**Breaking Changes**

<!-- Does this PR contain any backward incompatible changes? If so, what are they and should there be special considerations for release? -->

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [ ] No - this PR is backwards compatible

**Features**:

- Details ...

**Improvements**:

- Details ...

**Bug Fixes**:

- Details ...

## Before & After Screenshots

**BEFORE**:

<!-- [insert screenshot here] -->

**AFTER**:

<!-- [insert screenshot here] -->

## Tests

<!-- What tests should be run to confirm functionality? -->

## Deploy Notes

<!-- Notes regarding deployment of the contained body of work.  -->
<!-- These should note any new dependencies, new scripts, etc. -->

**New environment variables**:

- `env var` : env var details

**New scripts**:

- `script` : script details

**New dependencies**:

- `dependency` : dependency details

**New dev dependencies**:

- `dependency` : dependency details

alexanderleegs reviewed (commented):
The refactored workspace looks great, thanks for all the work on this! Have a few minor comments.

For Section/SectionCaption/SectionHeader, what do you think about consolidating these under their own separate section (maybe `components/sections` or something), so that it's easy to find grouped components? `CreateButton` could be renamed and go in here too if this is specifically used for section creation.

The other issue is a display related one for multi-line cards:
<img width="1256" alt="Screenshot 2022-06-09 at 2 23 20 PM" src="https://user-images.githubusercontent.com/22111124/172778761-e6942469-fcbd-42ca-83c1-3d13b4335534.png">

I see the figma does require multi-line cards, but the context menu button for these cases seem to be off (see the second row of cards). I was also wondering if we were okay with uneven positioning of the elements if the length of titles are different.

seaerchin commented:
> The refactored workspace looks great, thanks for all the work on this! Have a few minor comments.
> 
> For Section/SectionCaption/SectionHeader, what do you think about consolidating these under their own separate section (maybe `components/sections` or something), so that it's easy to find grouped components? `CreateButton` could be renamed and go in here too if this is specifically used for section creation.
> 
> The other issue is a display related one for multi-line cards: <img alt="Screenshot 2022-06-09 at 2 23 20 PM" width="1256" src="https://user-images.githubusercontent.com/22111124/172778761-e6942469-fcbd-42ca-83c1-3d13b4335534.png">
> 
> I see the figma does require multi-line cards, but the context menu button for these cases seem to be off (see the second row of cards). I was also wondering if we were okay with uneven positioning of the elements if the length of titles are different.

for the first point, yea, i did it in folders HAHA - i think it's worth relooking at how we want to structure our components but ideally we'd group them by "features"

for the second one - good catch, i actually didn't realise this! i think i need to ask @NatMaeTan for her input - this occurs because we resize the cards (for the grid) but the buttons are absolutely positioned wrt the cards so only hte affected card has proper sizing.

we could resize the whole row of cards or to simply cut text off at 1 line but i'm unsure which is best

alexanderleegs reviewed (commented):
Actually another thing I noticed - the sidebar is now scrollable and scrolls with the page, when it should be fixed

alexanderleegs reviewed (commented):
E2E tests are also failing due to the refactor - will chat with Kathleen before deciding whether to push this out before tests are fixed (we would have to run through our manual suite)

seaerchin commented:
@alexanderleegs gonna consolidate replies to avoid spam HAHA
> the sidebar is now scrollable and scrolls with the page, when it should be fixed

fixed in folders PR

> I see the figma does require multi-line cards, but the context menu button for these cases seem to be off (see the second row of cards). I was also wondering if we were okay with uneven positioning of the elements if the length of titles are different.

fixed - clarified with @NatMaeTan on resize vs cutoff; for multiline cards, restrict to 3 lines of text and for single, 1 line.

alexanderleegs reviewed (approved):
lgtm
